src/main/scala/com/example/TaskActor.scala

```diff
@@ -0,0 +1,245 @@
+package com.example
+
+import akka.NotUsed
+import akka.actor.typed.scaladsl.{ActorContext, Behaviors}
+import akka.actor.typed.{ActorRef, Behavior}
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.client.RequestBuilding.Get
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.model.StatusCodes.OK
+import akka.stream._
+import akka.stream.alpakka.csv.scaladsl.{CsvParsing, CsvToMap}
+import akka.stream.scaladsl.{FileIO, Flow, Keep, Sink, Source}
+import akka.util.ByteString
+import io.circe.Codec
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.semiauto.deriveEnumerationCodec
+import spray.json.{JsValue, JsonWriter}
+
+import java.nio.file.Paths
+import scala.collection.immutable
+import scala.collection.immutable.Queue
+import scala.language.postfixOps
+import scala.util.{Failure, Success}
+
+
+/**
+ * @author Petros Siatos
+ */
+object TaskActor {
+
+  def jsonUri(taskId: Int): String = s"localhost:8080/files/$taskId"
```

Comment:
URL `scheme` part is missing. If you didn't want to have this configured in the config, imagine multiple ways how to reach the service under different domains, what could be used instead?